core-web/libs/block-editor/src/lib/extensions/action-button/actions-menu.extension.ts

```diff
@@ -165,12 +173,38 @@ function execCommand({
         video: () => editor.commands.openAssetForm({ type: 'video' })
     };
 
+    getCustomActions(customBlocks).forEach((option) => {
+        whatToDo[option.id] = () => {
+            try {
+                editor.commands[option.commandKey]();
+            } catch {
+                console.warn(`Custom command ${option.commandKey} does not exists.`);
+            }
+        };
+    });
+
     whatToDo[props.type.name]
         ? whatToDo[props.type.name]()
         : editor.chain().setTextSelection(range).focus().run();
 }
 
-export const ActionsMenu = (viewContainerRef: ViewContainerRef) => {
+function mapCustomActions(actions): Array<DotMenuItem> {
+    return actions.map((action) => ({
+        icon: action.icon,
+        label: action.menuLabel,
+        commandKey: action.command,
+        id: action.name
+    }));
+}
+
+function getCustomActions(customBlocks): Array<DotMenuItem> {
+    return flatten(customBlocks.extensions.map((extension) => mapCustomActions(extension.actions)));
```

Comment:
I don't think we need lodash anymore: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat

Also, not sure if lodash has tree shaking in newer version, it used to be a problem back in the days.